.github/workflows/main.yml

```diff
@@ -0,0 +1,273 @@
+  test:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:10.1
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          # Maps tcp port 5432 on service container to the host
+          - 5432:5432
+    steps:
+      - name: Check out repository
+        uses: actions/checkout@v2
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Cache pip dependencies
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-pip-tests-warehouse-p
+        with:
+          path: ./.cache/pip
+          key: ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-
+
+      - name: Cache node version manager
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-nvm-dh
+        with:
+          path: ~/.nvm/versions/
+          key: ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-npm-modules-dh
+        with:
+          path: |
+            ~/.npm
+            **/node_modules
+          key: ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-${{ env.cache-name }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-${{ github.job }}-
+            ${{ runner.os }}-build-${{ env.CACHE_SEED }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+
+      - name: Install dependencies
+        run: sudo apt -y install libcurl4-openssl-dev libssl-dev pkg-config
+
+      - name: Install pip requirements
+        run: |
+          mkdir ./.cache && mkdir ./.cache/pip
+          chown -R $(whoami) ./.cache/pip
+          pip install -U pip setuptools wheel --cache-dir ./.cache/pip
+          pip install -r requirements/main.txt --cache-dir ./.cache/pip
+          pip install -r requirements/tests.txt --cache-dir ./.cache/pip
+          pip install -r requirements/deploy.txt --cache-dir ./.cache/pip
```

Comment:
Is there a reason we can't use the default (`~/.cache/pip`) here?

Reply:
I'll test this out with the default `~/.cache/pip` now that we're using the regular `ubuntu-latest` container, but I was having issues in the multi-stage Docker build where the permissions on `~/.cache/pip` were set such that I wasn't able to read or write to it.